crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs

```diff
@@ -304,3 +295,105 @@ pub(crate) fn filter_contained(edits: Vec<Edit>) -> Vec<Edit> {
     }
     filtered
 }
+
+#[derive(Copy, PartialEq, Clone)]
+enum State {
+    Literal,
+    AnnotatedFirst,
+    AnnotatedRest,
+    Other,
+}
+
+pub(crate) struct QuoteAnnotation<'a> {
+    state: Vec<State>,
+    stylist: &'a Stylist<'a>,
+    annotation: String,
+    final_quote_type: Quote,
+}
```

Comment:
nit: we can avoid storing the `final_quote_type` as that can be queried from `stylist.quote()` wherever required